source/agora/common/Config.d

```diff
@@ -330,6 +334,13 @@ private NodeConfig parseNodeConfig (Node* node, const ref CommandLine cmdln)
     auto min_listeners = get!(size_t, "node", "min_listeners")(cmdln, node);
     auto max_listeners = get!(size_t, "node", "max_listeners")(cmdln, node);
     auto address = get!(string, "node", "address")(cmdln, node);
+    auto commons_budget = opt!(string, "node", "commons_budget_address")(cmdln, node);
+
+    import agora.utils.WellKnownKeys;
+    auto commons_budget_address = (commons_budget.length > 0)
+        ? PublicKey.fromString(commons_budget)
+        : CommonsBudgetKeyPair.address;
```

Comment:
https://github.com/bpfkorea/agora/pull/1232#pullrequestreview-494067435

Reply:
This type of logic belongs somewhere here: https://github.com/bpfkorea/agora/blob/59d9463500e81f279e1ef57861d64246bf6e4e3b/source/agora/node/FullNode.d#L131

Then the budget address would be:

```D
auto commons_budget = 
    config.node.commons_budget_address != PublicKey.init ? 
    config.node.commons_budget_address : CNG.CommonsBudget;
```

And you would need to define a (fake) PublicKey inside of `agora.consensus.data.genesis.Coinnet`.

@Geod24 am I in the right here?

Reply:
I'll change it according to @AndrejMitrovic 's suggestion.